### ISSUE TITLE
RWK matrix returns by fit_transform by RandomWalkLabeled() contains only 1.0

### ISSUE THREAD

**Issue by KatrinaLiang** [ysig/GraKeL#96]
I'm using RandomWalkLabeled() to compute random walk kernel of node-labeled graphs with maximum step = 10 and different lambda values, as following:
```
lambdas = [0.0001, 0.001, 0.01, 0.1, 0.2, 0.3, 0.4, 1.2]

# Read control flow graphs from dot files with nx.drawing.nx_pydot.read_dot
cfg = np.asarray(read_dot_as_nxGraph(paths_to_dots), dtype=object)

#Transform networkx graphs to grakel representations
G_train = list(graph_from_networkx(cfg, node_labels_tag='label'))

for lambda_value in lambdas:
    rwk_kernel = RandomWalkLabeled(lamda=lambda_value, normalize=True, p=10)
    rwk_matrix = rwk_kernel.fit_transform(G_train)
```
The graphs have real-valued node labels, but no edge labels. Here is an example of my graphs:
```
G2.nodes(data=True)
-> NodeDataView({'0': {'style': 'filled', 'fillcolor': 'gray', 'label': 2}, '1': {'label': 27}, '2': {'label': 27}, '3': {'label': 27}, '4': {'label': 1}, '5': {'label': 27}, '10': {'style': 'filled', 'fillcolor': 'lightgray', 'label': 5}, '6': {'label': 6}, '7': {'label': 8}, '8': {'label': 8}, '9': {'label': 3}})

G2.edges(data=True)
-> OutMultiEdgeDataView([('0', '1', {}), ('1', '2', {}), ('2', '3', {}), ('3', '4', {}), ('4', '5', {}), ('4', '10', {}), ('5', '6', {}), ('6', '7', {}), ('7', '8', {}), ('8', '9', {}), ('9', '3', {})])
```

The problem is that every RWK matrix contains only 1.0. With normalize=False, the values in the matrix range from ~100 to ~2600, and they are the same for every lambda.

Could you spot anything wrong?
Thank you!

**Comment by giannisnik** [comment]
Hi @KatrinaLiang ,

This might depend on the dataset. Could you share a pair of graphs with us (edgelists of the two graphs along with node labels)?

**Comment by KatrinaLiang** [comment]
Thank you for your reply! @giannisnik 
Would it be okay if I share the whole dataset with you?
My dataset consists of 100 java programs in .dot format. You can see them in the dotFiles folder in [this repo](https://github.com/KatrinaLiang/RWKLabeled). The functions I used to read the dot files and give them to Grakel can be found in compute_rwk.py

**Comment by giannisnik** [comment]
@KatrinaLiang first of all, this implementation of the random walk kernel assumes discrete node features. If your features are continuous, there are kernels that can better capture the similarities of such features.

With regards to your question above, it turns out that if you initialize the kernel as follows: `rwk_kernel = RandomWalkLabeled(lamda=lambda_value, normalize=True)` and you don't specify the number of steps (which means that the kernel will compare walks of infinite length extracted from graphs), then the kernel values between structurally different graphs are smaller than 1.

If you specify the number of steps, all the kernel values are indeed equal to 1. This is not what I would expect the kernel to output. We will fix this bug in the next version. Thanks a lot for sharing this issue with us!

**Comment by KatrinaLiang** [comment]
@giannisnik Thank you!

I'm unsure about the concept of discrete node features; google didn't give me a lot of information on the term "dissect node". Are the nodes in my graphs discrete?

I would like to use Grakel for my master thesis, what is your plan on a new version?

**Comment by ysig** [comment]
Now this should be input to the labelled random walk kernel. However the adjacency matrix should be just the connections with 0 and 1 and the labels should be provided externally. The only time you want to have non binary values on your adjacency matrix is if they are non-negative floats and there the Random Walk one may give you something interesting.
Also for now just set p=None which is the default value.
Please read our documentation and examples and let us know if it's still unclear.
